cpp/open3d/t/geometry/BoundingVolume.cpp

```
@@ -289,6 +324,346 @@ AxisAlignedBoundingBox AxisAlignedBoundingBox::FromLegacy(
+    const core::Tensor x_axis =
+            rotation_.GetItem(
+                    {core::TensorKey::Slice(core::None, core::None, core::None),
+                     core::TensorKey::Index(0)}) *
+            extent_.GetItem({core::TensorKey::Index(0)}) * 0.5;
+    const core::Tensor y_axis =
+            rotation_.GetItem(
+                    {core::TensorKey::Slice(core::None, core::None, core::None),
+                     core::TensorKey::Index(1)}) *
+            extent_.GetItem({core::TensorKey::Index(1)}) * 0.5;
+    const core::Tensor z_axis =
+            rotation_.GetItem(
+                    {core::TensorKey::Slice(core::None, core::None, core::None),
+                     core::TensorKey::Index(2)}) *
+            extent_.GetItem({core::TensorKey::Index(2)}) * 0.5;
+
+    core::Tensor points = core::Tensor::Zeros({8, 3}, GetDtype(), GetDevice());
+    points.SetItem(core::TensorKey::Index(0),
+                   center_ - x_axis - y_axis - z_axis);
+    points.SetItem(core::TensorKey::Index(1),
+                   center_ + x_axis - y_axis - z_axis);
+    points.SetItem(core::TensorKey::Index(2),
+                   center_ - x_axis + y_axis - z_axis);
+    points.SetItem(core::TensorKey::Index(3),
+                   center_ - x_axis - y_axis + z_axis);
+    points.SetItem(core::TensorKey::Index(4),
+                   center_ + x_axis + y_axis + z_axis);
+    points.SetItem(core::TensorKey::Index(5),
+                   center_ - x_axis + y_axis + z_axis);
+    points.SetItem(core::TensorKey::Index(6),
+                   center_ + x_axis - y_axis + z_axis);
+    points.SetItem(core::TensorKey::Index(7),
+                   center_ + x_axis + y_axis - z_axis);
+    return points;
+}
+
+OrientedBoundingBox &OrientedBoundingBox::Translate(
+        const core::Tensor &translation, bool relative) {
+    core::AssertTensorDevice(translation, GetDevice());
+    core::AssertTensorShape(translation, {3});
+    if (translation.GetDtype() != GetDtype()) {
```

Skip warning, silently convert.